trieste/models/gpflow/models.py

```diff
@@ -601,27 +672,67 @@ def __init__(
 
         self._use_natgrads = use_natgrads
         self._natgrad_gamma = natgrad_gamma
+        self._ensure_variable_model_data()
+
+    def _ensure_variable_model_data(self) -> None:
```

Comment:
this whole check could potentially live in util.py as well, takes a single argument (model) and optionally returns three objects with which we update some parts of it